notifier/senders/mail/mail.go

```diff
@@ -30,18 +31,24 @@ type Sender struct {
 	dateTimeFormat string
 }
 
-// Init read yaml config
-func (sender *Sender) Init(senderSettings map[string]string, logger moira.Logger, location *time.Location, dateTimeFormat string) error {
+// NewSender creates Sender instance.
+func NewSender(senderSettings map[string]string, logger moira.Logger, location *time.Location, dateTimeFormat string) (*Sender, error) {
```

Comment:
Maybe NewSender methods should return interfaces instance instead of structure instance, so that we could hide the implementation inside the package

Reply:
It is interesting, I like your idea, so caller will have access only to interface methods and can't access to other methods and struct fields, but it breaks some tests (at least mattermost/internal_test.go). So I think if do that refactor, then need to refactor all Senders, but now it breaks, so it is not one step refactor and should be done in another PR. And I found that returning interface is controversial, discussion https://stackoverflow.com/questions/59040392/how-to-write-idiomatic-constructor and rule "accepting interfaces and returning structs" https://tutorialedge.net/golang/accept-interfaces-return-structs/

Reply:
I believe in this case it woukld be acceptable to return interface because:
1) It is not a library with a public api
2) The only thing we do with the return value is that we assign it to a variable with an interface type

On the other hand, there is no point in hiding implementation precisely because it is not a library with a public api :-)

So I am resolving the issue